### ISSUE TITLE
specs2 exception with no stack trace (IntelliJ 14.1.1, specs2 3.7)

### ISSUE THREAD

**Issue by remster** [etorreborre/specs2#455]
```
Testing started at 09:55 ...

This looks like a specs2 exception...
Please report it with the preceding stacktrace at http://github.com/etorreborre/specs2/issues
 
Process finished with exit code 0
```
IntelliJ could be at fault here. I can't post the faulty test spec right now.

**Comment by etorreborre** [comment]
I am really sorry that you got this, and I would like to understand why you don't have a corresponding stacktrace. I would appreciate if you could create a small specification reproducing the issue.

You could check if there are `vals` in your specification throwing exceptions, like
```scala
class MySpec extends Specification {
  val a = { sys.error("boom"); 1 }

  "my spec doesn't work" >> ok
}
```

**Comment by remster** [comment]
Hey Eric.,

No worries. I can usually work around these annoyances. You asked for a bug
report, I thought that was a small favour.

The snippet you provided behaves the same way as the spec that reproduces
the problem. Indeed an override val is involved,
but I can't find evidence that it throws.

I tried distilling it to something i can share with you - but so far can't
find the combination that reproduces the problem.
I'll retry in Monday.

Thanks for reacting so quickly.
Have a good weekend
Remek

**Comment by oswaldo** [comment]
Hello,

Please allow me to collaborate with a couple of cases which make this issue specially confusing. In real scenarios it is common that multiple APIs and traits get into the mix, making it particularly hard to debug if the actual exceptions are swallowed.
In the cases below I expect to see the originating exception somehow in the logs, not a filtered or wrapping stuff:

```scala
import org.junit.runner.RunWith
import org.specs2.mutable.Specification
import org.specs2.runner.JUnitRunner

@RunWith(classOf[JUnitRunner])
class BrokenTest extends Specification with BreakingTrait {

  "will fail" in {
    success
  }

  //Exception:
  //java.lang.ClassNotFoundException: BrokenTest$
  //	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
  //	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
  //	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
  //	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
  //	at org.specs2.reflect.Classes$$anonfun$loadClassEither$1.apply(Classes.scala:103)
  //	at org.specs2.reflect.Classes$$anonfun$loadClassEither$1.apply(Classes.scala:103)
  //	at scalaz.Name$$anon$5.value(Name.scala:18)
  //	at org.specs2.control.eff.ErrorEffect$$anon$1.apply(ErrorEffect.scala:60)
  //	at org.specs2.control.eff.ErrorEffect$$anon$1.apply(ErrorEffect.scala:53)
  //	at org.specs2.control.eff.Interpret$$anon$1.onEffect(Interpret.scala:55)
  //	at org.specs2.control.eff.Interpret$$anon$1.onEffect(Interpret.scala:47)
  //	at org.specs2.control.eff.Interpret$class.go$1(Interpret.scala:134)
  //	at org.specs2.control.eff.Interpret$class.interpretLoop(Interpret.scala:145)
  //	at org.specs2.control.eff.Interpret$.interpretLoop(Interpret.scala:153)
  //	at org.specs2.control.eff.Interpret$class.interpret(Interpret.scala:57)
  //	at org.specs2.control.eff.Interpret$.interpret(Interpret.scala:153)
  //	at org.specs2.control.eff.Interpret$class.interpret1(Interpret.scala:64)
  //	at org.specs2.control.eff.Interpret$.interpret1(Interpret.scala:153)
  //	at org.specs2.control.eff.ErrorEffect$class.runError(ErrorEffect.scala:65)
  //	at org.specs2.control.eff.ErrorEffect$.runError(ErrorEffect.scala:120)
  //	at org.specs2.control.package$.attemptExecuteAction(package.scala:64)
  //	at org.specs2.control.package$.runAction(package.scala:59)
  //	at org.specs2.runner.JUnitRunner.specification$lzycompute(JUnitRunner.scala:18)
  //	at org.specs2.runner.JUnitRunner.specification(JUnitRunner.scala:17)
  //	at org.specs2.runner.JUnitRunner.specStructure$lzycompute(JUnitRunner.scala:28)
  //	at org.specs2.runner.JUnitRunner.specStructure(JUnitRunner.scala:28)
  //	at org.specs2.runner.JUnitRunner.getDescription$lzycompute(JUnitRunner.scala:34)
  //	at org.specs2.runner.JUnitRunner.getDescription(JUnitRunner.scala:34)
  //...
}

trait BreakingTrait {
  throw new RuntimeException("this should not be swallowed")
}

object ExceptionShouldBeDisplayed {

  def main(args: Array[String]): Unit = {

    specs2.run.run(Array(classOf[BrokenTest].getName))

  }
  //Output:
  //This looks like a specs2 exception...
  //Please report it with the preceding stacktrace at http://github.com/etorreborre/specs2/issues
}

@RunWith(classOf[JUnitRunner])
class AnotherBrokenTest extends Specification with AnotherBreakingTrait {

  "will also fail" in {
    pow
    success
  }

  //Exception:
  //java.lang.ClassNotFoundException: AnotherBrokenTest$
  //	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
  //	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
  //	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
  //	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
  //	at org.specs2.reflect.Classes$$anonfun$loadClassEither$1.apply(Classes.scala:103)
  //	at org.specs2.reflect.Classes$$anonfun$loadClassEither$1.apply(Classes.scala:103)
  //	at scalaz.Name$$anon$5.value(Name.scala:18)
  //	at org.specs2.control.eff.ErrorEffect$$anon$1.apply(ErrorEffect.scala:60)
  //	at org.specs2.control.eff.ErrorEffect$$anon$1.apply(ErrorEffect.scala:53)
  //	at org.specs2.control.eff.Interpret$$anon$1.onEffect(Interpret.scala:55)
  //	at org.specs2.control.eff.Interpret$$anon$1.onEffect(Interpret.scala:47)
  //	at org.specs2.control.eff.Interpret$class.go$1(Interpret.scala:134)
  //	at org.specs2.control.eff.Interpret$class.interpretLoop(Interpret.scala:145)
  //	at org.specs2.control.eff.Interpret$.interpretLoop(Interpret.scala:153)
  //	at org.specs2.control.eff.Interpret$class.interpret(Interpret.scala:57)
  //	at org.specs2.control.eff.Interpret$.interpret(Interpret.scala:153)
  //	at org.specs2.control.eff.Interpret$class.interpret1(Interpret.scala:64)
  //	at org.specs2.control.eff.Interpret$.interpret1(Interpret.scala:153)
  //	at org.specs2.control.eff.ErrorEffect$class.runError(ErrorEffect.scala:65)
  //	at org.specs2.control.eff.ErrorEffect$.runError(ErrorEffect.scala:120)
  //	at org.specs2.control.package$.attemptExecuteAction(package.scala:64)
  //	at org.specs2.control.package$.runAction(package.scala:59)
  //	at org.specs2.runner.JUnitRunner.specification$lzycompute(JUnitRunner.scala:18)
  //	at org.specs2.runner.JUnitRunner.specification(JUnitRunner.scala:17)
  //	at org.specs2.runner.JUnitRunner.specStructure$lzycompute(JUnitRunner.scala:28)
  //	at org.specs2.runner.JUnitRunner.specStructure(JUnitRunner.scala:28)
  //	at org.specs2.runner.JUnitRunner.getDescription$lzycompute(JUnitRunner.scala:34)
  //	at org.specs2.runner.JUnitRunner.getDescription(JUnitRunner.scala:34)
  //...
}

trait AnotherBreakingTrait {
  val pow = throw new RuntimeException("this also should not be swallowed")
}

object ExceptionShouldAlsoBeDisplayed {

  def main(args: Array[String]): Unit = {

    specs2.run.run(Array(classOf[AnotherBrokenTest].getName))

  }
  //Output:
  //This looks like a specs2 exception...
  //Please report it with the preceding stacktrace at http://github.com/etorreborre/specs2/issues
}
```

The workaround for now is to create a main method that directly calls the suspicious code until you see the root cause so you are able to fix it.

**Comment by etorreborre** [comment]
Thanks for the detailed example, I am going to have a look at it.

**Comment by etorreborre** [comment]
The fix is available in `3.7-20160219163315-d9c5826`. Can you please confirm that it works for you?

**Comment by oswaldo** [comment]
That is great. Thanks for the prompt response.
